main.py

```diff
@@ -22,10 +21,19 @@ def load_data(path,startrows=None,endrows=None,cols=None):
     # endRow – last row to load, or the last of file, the minimum
     # cols – columns to load, there may be columns we would not want to load as they are useless
     # data – return the loaded partial file
-    if path[-5:] == ".xlsx" or path[-4:] == ".xls":
-        temp = pd.read_excel(path,usecols=cols,nrows=endrows-startrows,skiprows=startrows)
-    elif path[-4:] == ".csv":
-        temp=pd.read_csv(path,usecols=cols,nrows=endrows-startrows,skiprows=startrows)
+    row_num = endrows-startrows if startrows else endrows
+    try:
+        if path[-5:] == ".xlsx" or path[-4:] == ".xls":
+            header =pd.read_excel(path,usecols=cols,nrows=1)
+            temp = pd.read_excel(path,usecols=cols,nrows=row_num,skiprows=startrows)
+            temp.columns = header.columns
+        elif path[-4:] == ".csv":
+            header = pd.read_csv(path, usecols=cols, nrows=1)
+            temp=pd.read_csv(path,usecols=cols,nrows=row_num,skiprows=startrows)
```

Comment:
[STYLE] space after the `=`

Reply:
Please read whatever style guide for Python you can find, for example this: https://peps.python.org/pep-0008/ and follow it.

Also, do you work with an IDE? If yes - make it check those kinds of stuff. If no - start working with an IDE